test/unit/communication/test_comm_sender.cpp

```diff
@@ -33,37 +39,33 @@ void test_exec() {
   int recv_rank = (rank != 0) ? rank - 1 : nprocs - 1;
   int tag = 0;
 
-  auto send_fut = pika::async(ex, MPI_Isend, send_buf.data(), size, dtype, send_rank, tag, comm);
-  auto recv_fut = pika::async(ex, MPI_Irecv, recv_buf.data(), size, dtype, recv_rank, tag, comm);
-  pika::wait_all(send_fut, recv_fut);
+  auto send = just(send_buf.data(), size, dtype, send_rank, tag, comm) | transform_mpi(MPI_Isend);
```

Comment:
```suggestion
  auto send = just(send_buf.data(), size, dtype, send_rank, tag, comm) | dlaf::internal::transformMPI(MPI_Isend);
```
?

Reply:
What is the difference?

Reply:
`dlaf::internal::transformMPI` is the DLAF helper adaptor which does unwrapping and, for the moment, does `yield_while` on the MPI request. I.e. this is what is currently actually used for communication on this branch. `pika::mpi::experimental::transform_mpi` is the pika facility that does the same with scheduler polling but no unwrapping. This is not currently used because of the performance drop that I mentioned earlier.

Eventually I want DLAF's wrapper `transformMPI` to call `transform_mpi` (potentially with unwrapping if required). Since `transformMPI` will be around in the future even if the implementation changes I'm just asking myself if this should use `transformMPI`. I'm 99% certain the answer is yes.